modules/principals/src/main/frontend/src/Userboard/Groups/addusertogroupdialogue.jsx

```diff
@@ -0,0 +1,187 @@
+            })
+            .then((data) => {
+              var groupUsers = [];
+              var i;
+              for (i = 0; i < data.members.length; ++i) {
+                let username = data.members[i];
+                username = username.split('/').pop();
+                groupUsers.push(username);
+              }
+
+              let filtered = [];
+              filtered = this.state.allUserNames.filter( function( el ) {
+                  return !groupUsers.includes( el.name );
+                } );
+              this.setState({ freeUserNames: filtered });
+            })
+            .catch((error) => {
+              console.log(error);
+            });
+        }
+    }
+
+    handleExit() {
+        this.setState({ selectedUsers: [] });
+        this.state.allUserNames.forEach(function(item, index) {
+          item.tableData.checked = false;
+        })
+        this.props.handleClose();
+    }
+
+    render() {
+        return (
+            <Dialog
+                maxWidth="sm"
+                open={this.props.isOpen}
+                onEntering={() => this.handleEntering()}
+                onClose={() => this.handleExit()}
+            >
+                <DialogTitle>
+                    Add Users to the {this.props.name} group
+                </DialogTitle>
+                <DialogContent>
+                    <Grid container>
+                        <div>
+                            <MaterialTable
+                              title=""
+                              style={{ boxShadow : 'none' }}
+                              options={
+                                { draggable: false },
+                                { headerStyle: { backgroundColor: '#fafbfc'} },
+                                { emptyRowsWhenPaging: false }
+                              }
+                              columns={[
+                                { title: 'User Name', field: 'name' },
+                              ]}
+                              data={this.state.freeUserNames}
+                              options={{
+                                selection: true,
+                                showSelectAllCheckbox : false,
+                                showTextRowsSelected: false,
+                                selectionProps: rowData => ({
+                                    color: 'primary'
+                                  })
+                              }}
+                              onSelectionChange={(rows) => {this.handleSelectRowClick(rows)}}
+                            />
+                        </div>
+                    </Grid>
+                </DialogContent>
+                <DialogActions>
+                    <Button variant="contained" size="small" color="primary" onClick={() => this.handleAddUsers()} disabled={this.state.selectedUsers.length == 0}>Add</Button>
+                    <Button variant="contained" size="small" onClick={() => this.handleExit()}>Close</Button>
```

Comment:
Arrow function shouldn't be necessary here
```suggestion
                    <Button variant="contained" size="small" onClick={this.handleExit}>Close</Button>
```